docs/source/examples/dataframe-hdf5.rst

```diff
@@ -0,0 +1,111 @@
+
+
+**Generate Synthetic HDF5 files:**
+--------------------------------------------
+
+These exercises require multiple HDF5 files with multiple dataset keys.  The `Python pandas library <http://pandas.pydata.org/pandas-docs/stable/>`_ has a convenient `I/O API to create an HDF5 file from a dataframe <http://pandas.pydata.org/pandas-docs/stable/io.html#io-hdf5>`_. Combining this method with the `Python numpy random integer generator <http://docs.scipy.org/doc/numpy/reference/generated/numpy.random.randint.html>`_ and several standard library modules provide all the tools necessary to create a synthetic dataset.
+
+.. code-block:: python
+   :emphasize-lines: 40,45
+   
+    
+    def create_hdf5s(iters):
+           
+       import random
+       import pandas as pd
+       import numpy as np
+       import string 
+       import json
+    
+       # Intitalize counter to build specific number of hdf5 files
+       count = 0
+       setseed=1
+       np.random.seed(seed=setseed)
+       random.seed(setseed)
+    
+       # Key to store group keys and file names for generated hdf5 files
+       fileKeys = {}
+    
+       while count <= iters:
+            
+           # randomly pick letter as dataset key
+           groupkey = random.choice(list(string.ascii_lowercase))
+        
+           # randomly pick a number as hdf5 filename
+           saver = 'my'+str(np.random.randint(100))+'.h5'
+        
+           # Make a dataframe; 26 rows, 2 columns
+           df = pd.DataFrame(
+           {'x': np.random.randint(1,1000,26),
+           'y': np.random.randint(1,1000,26)},
+           index=list(string.ascii_lowercase)
+           )
+        
+           # Write synthetic hdf5 to current directory
+           df.to_hdf(saver, key='/'+groupkey, format='table')
+           fileKeys[saver]=groupkey
+           count += 1
+           setseed+=1
+        
+       # Return the dictionary with file key and filename for testing
+       with open('hdf5Keys.json','w') as f:
+           f.write(json.dumps(fileKeys))
+
+    return fileKeys
+
+    # Create the HDF5 files and keys to access
+    hdf5Keys = create_hdf5s(keys)
+
+        
+**Read single dataset from HDF5:**
+--------------------------------------------
+
+The first order of ``data.dataframe`` business is creating a dask DataFrame using a single HDF5 file's dataset.  The code to accomplish this task is::
+
+    import dask.dataframe as dd
+    df = dd.read_hdf(path='my10.h5',key='/u')
+    df['x'].compute() 
+    
+    
+**Load multiple datasets from single HDF5 file:**
+-------------------------------------------- 
+
+Loading multiple datasets from a single file requires a small tweak and use of the wildcard charater.  After creating a dask DataFrame object, use the ``dask.dataframe.DataFrame.visualize`` method to `render the computation of the task graph using graphviz <http://dask.pydata.org/en/latest/dataframe-api.html?highlight=visualize#dask.dataframe.DataFrame.visualize>`_ ::
```

Comment:
I'm tempted to cut out the visualize part.  It's not particularly relevant to `dd.read_hdf`.  This would be good if this were a tutorial we were giving somewhere (although i guess this may be your intention) but feels somewhat unnecessary for a documentation page (which I think should be somewhat direct.)